sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -1001,7 +1001,13 @@ createOrReplaceTableColTypeList
     ;
 
 createOrReplaceTableColType
-    : colName=errorCapturingIdentifier dataType (NOT NULL)? defaultExpression? commentSpec?
+    : colName=errorCapturingIdentifier dataType colDefinitionOption*
```

Comment:
Sorry for the late comment:

Just to confirm, if this change extends the options support. Looks like in the past there is an ordering for those options (`NOT NULL` first, default values next, comment spec last) but with this PR, there is no such ordering anymore?

Reply:
No worries. Yes, this is correct. This PR relaxes the ordering constraint such that any combination of those options is considered valid. This makes it easier to apply the CREATE TABLE column options.